packages/jbrowse-plugin-apollo/src/ApolloInternetAccount/components/ApolloLoginForm.tsx

```diff
@@ -0,0 +1,102 @@
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  TextField,
+} from '@material-ui/core'
+import React, { useState } from 'react'
+
+export const ApolloLoginForm = ({
+  authenticationURL,
+  internetAccountId,
+  handleClose,
+}: {
+  authenticationURL: string
+  internetAccountId: string
+  handleClose: (arg?: string) => void
+}) => {
+  const [username, setUsername] = useState('')
+  const [password, setPassword] = useState('')
+
+  async function onSubmit(
+    this: unknown,
+    event: React.FormEvent<HTMLFormElement>,
+  ) {
+    if (username && password) {
+      const url = new URL(authenticationURL)
+      const paramsString = `username=${username}&password=${password}`
+      const queryParams = new URLSearchParams(paramsString)
+      url.search = queryParams.toString()
```

Comment:
It's probably better to have the username/password in the POST body instead of the URL, so it would look something like:

```js
const res = await fetch(authenticationURL, {
  method: 'POST',
  body: JSON.stringify({ username, password }),
  headers: { 'Content-Type': 'application/json' },
})
```

(In my testing, it needed the 'Content-Type' header to work properly.)

Reply:
Done as proposed